gravity-forms/gw-format-date-merge-tags.php

```diff
@@ -9,50 +9,96 @@
+		add_action( 'init', array( $this, 'init' ) );
+	}
 
-		$i        = $match[0][0] === '{' ? 4 : 5;
-		$modifier = rgar( array_map( 'trim', explode( ',', rgar( $match, $i ) ) ), 0 );
-		if ( ! $modifier ) {
-			continue;
-		}
+	public function init() {
+		add_filter( 'gform_pre_replace_merge_tags', array( $this, 'replace_merge_tags' ), 10, 7 );
+	}
 
-		$value = GFFormsModel::get_lead_field_value( $entry, $field );
-		$value = $field->get_value_merge_tag( $value, $input_id, $entry, $form, $modifier, $value, $url_encode, $esc_html, $format, $nl2br );
-		if ( ! $value ) {
-			continue;
-		}
+	public function replace_merge_tags( $text, $form, $entry, $url_encode, $esc_html, $nl2br, $format ) {
+		$locale = $this->_args['locale'];
+
+		preg_match_all( '/{[^{]*?:(\d+(\.\d+)?)(:(.*?))?}/mi', $text, $matches, PREG_SET_ORDER );
+
+		foreach ( $matches as $match ) {
+
+			$input_id = $match[1];
+			$field    = GFFormsModel::get_field( $form, $input_id );
+
+			if ( ! $field || $field->get_input_type() !== 'date' ) {
+				continue;
+			}
+
+			$i        = $match[0][0] === '{' ? 4 : 5;
+			$modifier = rgar( array_map( 'trim', explode( ',', rgar( $match, $i ) ) ), 0 );
+			if ( ! $modifier ) {
+				continue;
+			}
 
-		$format      = $field->dateFormat ? $field->dateFormat : 'mdy';
-		$parsed_date = GFCommon::parse_date( $value, $format );
+			$value = GFFormsModel::get_lead_field_value( $entry, $field );
+			$value = $field->get_value_merge_tag( $value, $input_id, $entry, $form, $modifier, $value, $url_encode, $esc_html, $format, $nl2br );
+			if ( ! $value ) {
+				continue;
+			}
 
-		// On the Notifications/Confirmation side, & gets encoded to &amp;. Decode it back.
-		$modifier = htmlspecialchars_decode( $modifier );
+			$format      = $field->dateFormat ? $field->dateFormat : 'mdy';
+			$parsed_date = GFCommon::parse_date( $value, $format );
 
-		// For whatever reason, Populate Anything's LMTs works better with `&comma` than `&#44;`. But... date() doesn't
-		// like it so let's replace it before we pass it to date().
-		$modifier = str_replace( '&comma;', ',', $modifier );
+			// On the Notifications/Confirmation side, & gets encoded to &amp;. Decode it back.
+			$modifier = htmlspecialchars_decode( $modifier );
 
-		// phpcs:ignore WordPress.DateTime.RestrictedFunctions.date_date
-		$replace = date( $modifier, strtotime( sprintf( '%d-%d-%d', $parsed_date['year'], $parsed_date['month'], $parsed_date['day'] ) ) );
+			// For whatever reason, Populate Anything's LMTs works better with `&comma` than `&#44;`. But... date() doesn't
+			// like it so let's replace it before we pass it to date().
+			$modifier = str_replace( '&comma;', ',', $modifier );
 
-		$text = str_replace( $match[0], $replace, $text );
+			$timestamp = strtotime( sprintf( '%d-%d-%d', $parsed_date['year'], $parsed_date['month'], $parsed_date['day'] ) );
 
+			if ( $locale ) {
+				switch_to_locale( $locale );
+			}
```

Comment:
I'd put a switch back to the other locale right before the return of this function, just to be safe.

```
$current_locale = determine_locale();
```